osu.Game/Overlays/KeyBinding/RestorableKeyBindingRow.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Input.Bindings;
+using osu.Game.Rulesets;
+
+namespace osu.Game.Overlays.KeyBinding
+{
+    public class RestorableKeyBindingRow : Container, IFilterable
+    {
+        private readonly object key;
+        private readonly ICollection<Input.Bindings.DatabasedKeyBinding> bindings;
+        public readonly KeyBindingRow KeyBindingRow;
+
+        private bool matchingFilter;
+
+        public bool MatchingFilter
+        {
+            get => matchingFilter;
+            set
+            {
+                matchingFilter = value;
+                this.FadeTo(!matchingFilter ? 0 : 1);
+            }
+        }
+
+        public bool FilteringActive { get; set; }
+
+        public IEnumerable<string> FilterTerms => bindings.Select(b => b.KeyCombination.ReadableString()).Prepend(key.ToString());
+
+        public RestorableKeyBindingRow(
```

Comment:
We don't format like this anywhere

Reply:
Here is one example where that format is used.

https://github.com/ppy/osu/blob/44b8725458a80c57b9c38337a1b7c42a91f2d226/osu.Game/Online/API/Requests/SearchBeatmapSetsRequest.cs#L47-L60

Reply:
I glanced over this as I actually tend to format code similarly with hanging indents if there are a lot of ctor params (i.e. constructor DI), but I can change to single line.